### PR TITLE
enable custom and multiple testnet deployment archive-node setups

### PR DESCRIPTION
**Pull request by 0x0I**
Update infrastructure components to allow for custom & multiple `archive-node` deployments including:
* updates to `o1-testnet` terraform module to support variable archive deployment configs 
* toggle of both a local Mina daemon and PostgresDB deployment alongside each archive server
* archival capabilities for all Mina daemons across testnet components/charts
* addition of db bootstrap job to support concurrent database transactions

Also ran `terraform` [fmt](https://www.terraform.io/docs/cli/commands/fmt.html) against affected testnet modules to promote consistent formatting.

**Test:** Buildkite CI

#7567

### CODE REVIEW

File: helm/archive-node/templates/genesis-configmap.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
-{{- if .Values.coda.runtimeConfig }}
+{{- if $.Values.coda.runtimeConfig }}
```
Comment (0x0I) [inline helm/archive-node/templates/genesis-configmap.yaml:1]: revert back to `.Values`

File: helm/archive-node/templates/genesis-configmap.yaml [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: "{{ template "archive-node.fullname" . }}-daemon-config"
   labels:
     app: {{ template "archive-node.fullname" . }}
-    heritage: {{ .Release.Service }}
-    release: {{ .Release.Name }}
+    heritage: {{ $.Release.Service }}
```
Comment (0x0I) [inline helm/archive-node/templates/genesis-configmap.yaml:8]: same here => `.Release`

File: helm/archive-node/values.yaml [hunk 3]
```diff
@@ -14,30 +14,31 @@ coda:
...
   image: codaprotocol/coda-archive:0.0.16-beta7-develop
-  listenPort: "3086"
-  nodeName: "dev"
   remoteSchemaFile: "https://raw.githubusercontent.com/MinaProtocol/mina/develop/src/app/archive/create_schema.sql"
-  postgresHost: '{{ .Release.Name }}-postgresql'
-  postgresPort: "5432"
+  postgresHost:  '{{ .Release.Name }}-postgresql'
   postgresDB: "archive"
-  postgresUri: postgres://{{ .Values.postgresql.postgresqlUsername }}:{{ .Values.postgresql.postgresqlPassword }}@{{ tpl .Values.archive.postgresHost . }}:{{ .Values.archive.postgresPort }}/{{ .Values.archive.postgresDB }}
+  postgresUri: postgres://{{ .Values.postgresql.postgresqlUsername }}:{{ .Values.postgresql.postgresqlPassword }}@archive-1-postgresql:{{ .Values.archive.ports.postgres }}/{{ .Values.archive.postgresDB }}
```
Comment (0x0I) [inline helm/archive-node/values.yaml:26]: revert back to `archive.postgresHost`

File: helm/archive-node/templates/_healthchecks.tpl [hunk 4]
```diff
@@ -28,7 +28,9 @@ readinessProbe:
     command: [
       "/bin/bash",
       "-c",
-      "source /healthcheck/utilities.sh && isDaemonSynced && isArchiveSynced --db-host {{ template "archive-node.fullname" . }}-postgresql"
+      "source /healthcheck/utilities.sh && isArchiveSynced \
```
Comment (0x0I) [inline helm/archive-node/templates/_healthchecks.tpl:31]: Only execute this readiness check if local daemon is enabled - TODO: check out replacing with archive-node auditor tool

File: automation/terraform/modules/kubernetes/testnet/variables.tf [hunk 5]
```diff
@@ -257,31 +252,43 @@ variable "make_report_accounts" {
...
+      size               = optional(string)
+      reclaimPolicy      = optional(string)
+      storageClass       = optional(string)
+      accessModes       = optional(list(string))
+  })
+  default = {
+    enabled = true
+    size  = "8Gi"
+    reclaimPolicy = "retain"
+    storageClass  = "us-west1-ssd-retain"
```
Comment (0x0I) [inline automation/terraform/modules/kubernetes/testnet/variables.tf:289]: rename storage class to remove redundant ${region} component e.g: us-west-1-ssd-retain => ssd-retain

File: helm/archive-node/values.yaml [hunk 6]
```diff
@@ -18,26 +18,25 @@ coda:
...
   image: codaprotocol/coda-archive:0.0.16-beta7-develop
-  listenPort: "3086"
-  nodeName: "dev"
   remoteSchemaFile: "https://raw.githubusercontent.com/MinaProtocol/mina/develop/src/app/archive/create_schema.sql"
-  postgresHost: '{{ .Release.Name }}-postgresql'
-  postgresPort: "5432"
+  postgresHost:  '{{ .Release.Name }}-postgresql'
   postgresDB: "archive"
-  postgresUri: postgres://{{ .Values.postgresql.postgresqlUsername }}:{{ .Values.postgresql.postgresqlPassword }}@{{ tpl .Values.archive.postgresHost . }}:{{ .Values.archive.postgresPort }}/{{ .Values.archive.postgresDB }}
+  postgresUri: postgres://{{ .Values.postgresql.postgresqlUsername }}:{{ .Values.postgresql.postgresqlPassword }}@{{ .Values.archive.postgresHost }}:{{ .Values.archive.ports.postgres }}/{{ .Values.archive.postgresDB }}
```
Comment (0x0I) [inline helm/archive-node/values.yaml:24]: read `tmp .Values.archive.postgresHost`

File: automation/terraform/modules/o1-testnet/inputs.tf [hunk 7]
```diff
@@ -211,7 +206,7 @@ variable "agent_send_every_mins" {
 }
 
 variable "gcloud_seeds" {
-  type    = list
+  type    = list(any)
```
Comment (0x0I) [inline automation/terraform/modules/o1-testnet/inputs.tf:209]: revert

File: automation/terraform/modules/o1-testnet/inputs.tf [hunk 8]
```diff
@@ -249,29 +244,14 @@ variable "make_report_accounts" {
...
+  type    = number
+  default = 0
 }
 
-variable "archive_persistence_size" {
-  type    = string
-  default = "8Gi"
+variable "archive_configs" {
+  type    = list(any)
+  default = []
```
Comment (0x0I) [inline automation/terraform/modules/o1-testnet/inputs.tf:254]: add description

File: automation/terraform/testnets/ci-net/main.tf [hunk 9]
```diff
@@ -82,43 +82,43 @@ module "ci_testnet" {
   artifact_path = var.ci_artifact_path
 
   # TODO: remove obsolete cluster_name var + cluster region
-  cluster_name          = "mina-integration-west1"
```
Comment (0x0I) [inline automation/terraform/testnets/ci-net/main.tf:85]: remove provider config above (represented by *ci_k8s_ctx*)

**Review by lk86 — APPROVED** [review]

**Review by imeckler — APPROVED** [review]
!approve-for-mainnet

**Pull request merged**